base/arrayshow.jl

```diff
@@ -320,14 +320,21 @@ print_array(io::IO, X::AbstractArray) = show_nd(io, X, print_matrix, true)
 # typeinfo aware
 # implements: show(io::IO, ::MIME"text/plain", X::AbstractArray)
 function show(io::IO, ::MIME"text/plain", X::AbstractArray)
+    # -1) handle arrays-in-array (recursive call)
+    if get(io, :compact, false) === true
+        # Using `show` of method with the most abstract type to avoid infinite recursion.
```

Comment:
Is this really necessary? I don't think it's common for 2-arg show methods to call 3-arg ones.